example/driver/simpledriver.go

```diff
@@ -48,6 +48,9 @@ type SimpleDriver struct {
 	serviceConfig        *config.ServiceConfig
 }
 
+// Do a runtime-check to determine that driver implements ProtocolDriver,
+var _ interfaces.ProtocolDriver = (*SimpleDriver)(nil)
```

Comment:
Why is this needed? I service will not compile if this is not the case.

Reply:
I add this like device-onvif-camera which can do runtime checking. https://github.com/edgexfoundry/device-onvif-camera/blob/main/internal/driver/driver.go#L89

Reply:
Ok the reason this is occurring is incorrect parameter typing on startup.Bootstrap which should in force the interface at compile time.

`func Bootstrap(serviceKey string, serviceVersion string, driver any)`
should be
`func Bootstrap(serviceKey string, serviceVersion string, driver interfaces.ProtocolDriver)`

Please fix it there and remove the check `protocolDriver, ok := driver.(interfaces.ProtocolDriver)` in `NewDeviceService`

Reply:
Refactored to use the specified interface instead of any type.